### ISSUE TITLE
Front end recording 

### ISSUE THREAD

**Issue by kgilpin** [getappmap/appmap-agent-js#227]
Record client side JavaScript running in the browser.

**Comment by ar-IGT** [comment]
Yup, especially with support for mainstream's most popular JS frameworks, like Angular, React, Vue and plain ES6/TypeScript ran on Webpack.

**Comment by yashwp** [comment]
This tool should be a must-integration for all frameworks!
Angular, Vue, Nuxt, Flutter, etc.

**Comment by ar-IGT** [comment]
Similar feature had JetBrains IntelliJ WebStorm and IDEA Ultimate, using [Spy.JS plugin](https://www.jetbrains.com/help/idea/2022.2/debugging-with-spy-js.html#0) that they developed, unfortunately [they decided to kill the project](https://youtrack.jetbrains.com/issue/WEB-64758/Cant-start-SpyJS-tracing-session-as-a-Debug-configuration-starting-as-Run-configuration-instead-but-SpyJS-tracing-doesnt-work-at).

Maybe @getappmap will be able to convince them to take over the plugin's codebase maintenance and merge it into AppMap JS Agent?

@kgilpin

**Comment by evlawler** [comment]
Thanks @ar-IGT and @yashwp for bringing up this project! I will reach out to them.  We are looking for ways to accelerate front to back end  recording support and this looks promising!
